### ISSUE TITLE
Static Site does not honor --domain option

### ISSUE THREAD

**Issue by cworkschris** [aws/copilot-cli#5159]
I tried this several times and then tried to simplify it.

My minimal example, which pushed up a bare helloworld html file:

```
 copilot app init newproj --domain something.net
 copilot svc init --app newproj --name front -t "Static Site"
 copilot env init
 copilot env deploy --name dev
 copilot svc deploy --name front
```

resulted in deployment to a cloud url:
```
...
  - You can access your service at https://dyd1ruk35v6vl.cloudfront.net/ over the internet.
...
```

I also tried this in an existing copilot app that was working with all services on a specified domain.

**Comment by KollaAdithya** [comment]
Hello @cworkschris !

Thanks for the issue on this enhancement.

Incase of other service type, We add an add an `A-record` with value `svcName.envName.appName.Domain` by default when we create an environment. And when a service is deployed we provide the url of `alias domain`.

In case of Static site We need to add an `A-record` with value `svcName.envName.appName.Domain`  to `environment hosted zone` by default if an app is associated with domain.

In this way we can provide the url with domain alias instead of cloudfront distribution endpoint.

**Comment by cworkschris** [comment]
Thank you @KollaAdithya!

That will meet user expectations I think - please keep up the great work, it really makes using fargate more pleasant!

**Comment by KollaAdithya** [comment]
Hello @cworkschris !

As a workaround can you modify you manifest to have

```
http:
  alias: <domain of the app>
```
In this way copilot will provide the URL of the domain alias.

**Comment by dannyrandall** [comment]
Hey @cworkschris! We just released this [v1.30.0](https://github.com/aws/copilot-cli/releases/tag/v1.30.0)!🚀🎉
